### PR TITLE
Feature/update with new args

### PR DESCRIPTION
**Pull request by rohitsinha54**
- Update dependencies
- Get arguments in spark with new api
- Use new unit tests methods.

### CODE REVIEW

File: MovieRecommender/src/test/java/co/cask/cdap/apps/movierecommender/MovieRecommenderAppTest.java [hunk 1]
```diff
@@ -59,10 +59,10 @@ public void testRecommendation() throws Exception {
 
     try {
       // Inject ratings data
-      sendRatingsData(appManager);
+      sendRatingsData();
       // Start the Spark Program
-      SparkManager sparkManager = appManager.startSpark(RecommendationBuilder.class.getSimpleName());
-      sparkManager.waitForFinish(120, TimeUnit.SECONDS);
+      SparkManager sparkManager = appManager.getSparkManager(RecommendationBuilder.class.getSimpleName()).start();
+      sparkManager.waitForFinish(60, TimeUnit.SECONDS);
```
Comment (ghost) [inline MovieRecommender/src/test/java/co/cask/cdap/apps/movierecommender/MovieRecommenderAppTest.java:65]: Why lower the wait time?

Comment (chtyim) [inline MovieRecommender/src/test/java/co/cask/cdap/apps/movierecommender/MovieRecommenderAppTest.java:65]: Because we run out of patience?

Comment (hsaputra) [inline MovieRecommender/src/test/java/co/cask/cdap/apps/movierecommender/MovieRecommenderAppTest.java:65]: touché =)

File: MovieRecommender/pom.xml [hunk 2]
```diff
@@ -66,8 +66,8 @@
       <scope>provided</scope>
       <exclusions>
         <exclusion>
-          <groupId>org.apache.hadoop</groupId>
-          <artifactId>hadoop-client</artifactId>
+          <groupId>com.sun.jersey</groupId>
```
Comment (hsaputra) [inline MovieRecommender/pom.xml:69]: Since the dependencies are provided, why do we need to add exclusion?

**Comment by hsaputra** [comment]
Did a pass and just have few questions but the rest look good to me.

File: MovieRecommender/src/test/java/co/cask/cdap/apps/movierecommender/MovieRecommenderAppTest.java (hunk 1 above)
Comment (rohitsinha54) [inline MovieRecommender/src/test/java/co/cask/cdap/apps/movierecommender/MovieRecommenderAppTest.java:65]: @chtyim :100:

File: MovieRecommender/pom.xml (hunk 2 above)
Comment (rohitsinha54) [inline MovieRecommender/pom.xml:69]: jersey brings in asm which collide with ours and during unit tests when this dependencies is brought in it makes our services fails as we get handler through reflection.

**Comment by rohitsinha54** [comment]
@henry-cask : Addressed comments. Have a look.

**Comment by ghost** [comment]
LGTM

**Pull request merged**